lib/model.js

```diff
@@ -340,7 +344,29 @@ ModelBaseClass.prototype._initProperties = function(data, options) {
       }
     }
   }
-  this.trigger('initialize');
+  this.__trigger('initialize');
+};
+
+ModelBaseClass.prototype.__trigger = function trigger(actionName, work, data, callback) {
```

Comment:
Yes please, let's remove the whole hook/trigger infrastructure and call `afterInitialize` (and/or `postInitialize`) directly.